tests/test_tarball.py

```diff
@@ -89,105 +89,30 @@ def test_version_configuration_override(self):
-        Case 3: Existent zip, but empty.
-        Case 4: Existent zip, with content and non common dir.
-        """
-
-        class MockZipFile:
-            def __init__(self, path, mode):
-                self.name = path
-                self.mode = mode
-                self.content = zip_content
-
-            def __enter__(self):
-                return self
-
-            def __exit__(self, exc_type, exc_val, traceback):
-                return False
-
-            def namelist(self):
-                return self.content
-
-        tests = (
-                ({'is_zip':True, 'content':['dir/','dir/file1']}, ('unzip -qq -d . path/zip_file-v1.zip', 'dir')),
-                ({'is_zip':False, 'content':[]}, (False, False)),
-                ({'is_zip':True, 'content':[]}, (False, False)),
-                ({'is_zip':True, 'content':['dir/','dir1/file1', ]}, ('unzip -qq -d ./zip_file-v1 path/zip_file-v1.zip', '')),
-                )
-
-        for input, expected in tests:
-            try:
-                tarball.print_fatal = mock_gen(rv=None)
-                tarball.zipfile.is_zipfile = mock_gen(rv=input['is_zip'])
-                zip_content = input['content']
-                tarball.build.base_path = '.'
-                tarball.zipfile.ZipFile = MockZipFile
-                actual = tarball.build_unzip('path/zip_file-v1.zip')
-            except:
-                actual = (False, False)
-            finally:
-                self.assertEqual(actual, expected)
-
     def test_find_extract(self):
         """
-        Test find_extract with the three supported filetypes
+        Test extract with the supported filetypes
         (*.zip, *.tar*)
         """
-        build_unzip_backup = tarball.build_unzip
-        build_untar_backup = tarball.build_untar
 
-        tarball.build_unzip = Mock(return_value=('', ''))
-        tarball.build_untar = Mock(return_value=('', ''))
+        build.base_path = 'tmpdir/tmp123'
+
+        get_zip_prefix_backup = tarball.Source.get_zip_prefix
+        get_tar_prefix_backup = tarball.Source.get_tar_prefix
+
+        tarball.Source.get_zip_prefix = Mock()
+        tarball.Source.get_tar_prefix = Mock()
+        tarball.Source.extract_zip = Mock()
+        tarball.Source.extract_tar = Mock()
 
-        tarball.find_extract('path/to/tar', 'test.zip')
-        tarball.find_extract('path/to/tar', 'test.tar.gz')
+        zip_tarball = tarball.Source('http://tarball.zip', '', 'tmpdir')
+        tar_tarball = tarball.Source('http://tarball.tar', '', 'tmpdir')
 
-        tarball.build_unzip.assert_called_once()
-        tarball.build_untar.assert_called_once()
+        tarball.Source.extract_zip.assert_called_once()
+        tarball.Source.extract_tar.assert_called_once()
 
-        tarball.build_unzip = build_unzip_backup
-        tarball.build_untar = build_untar_backup
+        tarball.Source.get_zip_prefix = get_zip_prefix_backup
+        tarball.Source.get_tar_prefix = get_tar_prefix_backup
```

Comment:
I think adding a test for ':' would make sense since that is a conditional branch. There are quite a few more conditionals that would also be nice to hit. I would like to get test coverage for your changes to be up near the 80% level.

Reply:
Already working on adding the tests, but would take me some to finish them :/

Reply:
This code being somewhat core to interacting with files, I'd like to ensure our coverage is healthy. I'm okay with seeing a set of tests that handle most of the scenarios (adding a case for archives/multiversion go seems helpful with these changes) rather than a strict coverage number.

Reply:
Just added more content with more scenarios. Most of the new content includes dynamic tests created on the fly, but also got some regular tests.